### PR TITLE
Pass agreementText in ProcessedDisclosedContract [DPP-1323]

### PR DESCRIPTION
**Pull request by tudor-da**
changelog_begin
changelog_end

### CODE REVIEW

File: daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala [hunk 1]
```diff
@@ -1746,6 +1748,7 @@ class EngineTest
           signatories = Set(alice),
           stakeholders = Set(alice),
           maybeKeyWithMaintainers = None,
+          agreementText = alice,
```
Comment (tudor-da) [inline daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala:1751]: @remyhaemmerle-da This test fails on an encoding issue. 
The agreement text function is essentially `show alice` which leads to `'Alice'` string literal as the returned agreement text in the cached contract. Is this expected? Does `show` enclose strings in quotes?

Comment (remyhaemmerle-da) [reply daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala:1751]:
```suggestion
          agreementText = s"'$alice'",
```

You can have a look at [DA.Internal.LF](https://github.com/digital-asset/daml/blob/8bcd24f9456b2b19a3e516d8add3ffa7233b480e/compiler/damlc/daml-stdlib-src/DA/Internal/LF.daml#L65) where the implementation of Show instance for Party is defined.

Comment (tudor-da) [reply daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala:1751]: Great, thanks for confirming.

Comment (tudor-da) [reply daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala]: @remyhaemmerle-da Changed. Can you please review?

**Review by remyhaemmerle-da — APPROVED** [review]
LGTM. Thanks.

**Comment by tudor-da** [comment]
Fixes https://github.com/digital-asset/daml/issues/15846

**Pull request merged**